x-pack/solutions/observability/packages/kbn-streams-schema/src/models/core.ts

```diff
@@ -33,6 +33,12 @@ export interface RecursiveRecord {
   [key: PropertyKey]: Primitive | Primitive[] | RecursiveRecord;
 }
 
-export const recursiveRecord: z.ZodType<RecursiveRecord> = z.record(
-  z.union([primitive, z.array(primitive), z.lazy(() => recursiveRecord)])
+export const recursiveRecord: z.ZodType<RecursiveRecord> = z.lazy(() =>
+  z.record(z.union([primitive, z.lazy(() => z.array(primitive)), recursiveRecord]))
+);
+
+export type FlattenRecord = Record<PropertyKey, Primitive | Primitive[]>;
+
+export const flattenRecord: z.ZodType<FlattenRecord> = z.record(
+  z.union([primitive, z.lazy(() => z.array(primitive))])
```

Comment:
this doesn't need to be lazy, correct?

Reply:
yep it's a left over from work in progress, good catch!